code/go/0chain.net/sharder/chain.go

```diff
@@ -206,8 +206,7 @@ func (sc *Chain) setupLatestBlocks(ctx context.Context, bl *blocksLoaded) (
 	}
 
 	sc.SetRandomSeed(bl.r, bl.r.GetRandomSeed())
-	bl.r.Block = bl.lfb
-	bl.r.BlockHash = bl.lfb.Hash
+	bl.r.Finalize(bl.lfb)
```

Comment:
why should we take care of finalization here, is it because we could crash in some inconsistent (partial) state last time and started with not round, which processing was not finished properly?

Reply:
It only set the block as finalized, nothing else actually. The reason to set it as finalized because otherwise the `if prevBlockHash == "" || !pr.IsFinalized() {` will fail for `round:1`.

Reply:
ok, got it